### PR TITLE
JetBrains: Cody: Add actions for cycling over autocomplete items

### LINKED ISSUE #55911: JetBrains: make it possible to cycle between multiple autocomplete items

### PR DESCRIPTION
**Pull request by Gedochao**
Fixes #55911 

https://github.com/sourcegraph/sourcegraph/assets/18601388/e82db13b-4a7f-4fac-958b-3b74055281a4

## Test plan
- when there's >1 autocomplete items available at a caret
  - `alt + [`/`alt + ]` should allow to cycle over the available items
  - pressing `tab` should apply the currently shown item
- when there's exactly 1 autocomplete item at a caret
  - `alt + [`/`alt + ]` should keep showing the same autocomplete item
- when there's no autocomplete items at a caret, the cycle actions shouldn't trigger at all (can trace it in debug logs)

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/cody/vscode/InlineAutocompleteItem.java [hunk 1]
```diff
@@ -83,4 +84,20 @@ public String toString() {
             : "";
     return new AutocompleteText(sameLineBeforeSuffixText, afterEndOfLineSuffix, blockText);
   }
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (szymonprz) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/vscode/InlineAutocompleteItem.java:89]: how about migrating this to kotlin data class?

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/autocomplete/action/AutocompleteActionHandler.kt [hunk 2]
```diff
@@ -21,7 +23,18 @@ open class AutocompleteActionHandler : EditorActionHandler() {
   }
 
   private fun hasAnyAutocompleteItems(project: Project, caret: Caret): Boolean =
-      CodyAgent.isConnected(project) && getAgentAutocompleteItem(caret) != null
+      CodyAgent.isConnected(project) && getCurrentAutocompleteItem(caret) != null
+
+  protected fun hasMultipleAutocompleteItems(project: Project, caret: Caret): Boolean {
+    val allAutocompleteItems = getAllAutocompleteItems(caret)
+    logger.warn("allAutocompleteItems: ${allAutocompleteItems.size}")
```
Comment (szymonprz) [inline client/jetbrains/src/main/kotlin/com/sourcegraph/cody/autocomplete/action/AutocompleteActionHandler.kt:30]: is this some kind of a leftover?

**Review by szymonprz — APPROVED** [review]

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/autocomplete/action/AutocompleteActionHandler.kt (hunk 2 above)
Comment (Gedochao) [reply client/jetbrains/src/main/kotlin/com/sourcegraph/cody/autocomplete/action/AutocompleteActionHandler.kt]: yup, removed it

File: client/jetbrains/src/main/java/com/sourcegraph/cody/vscode/InlineAutocompleteItem.java (hunk 1 above)
Comment (Gedochao) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/vscode/InlineAutocompleteItem.java]: done

**Review by olafurpg — APPROVED** [review]

**Pull request merged**